lib/transmit.js

```diff
@@ -24,32 +24,32 @@ exports.send = async function (request) {
             return;
         }
 
-        await internals.marshal(request);
+        await internals.marshal(response);
         await internals.transmit(response);
     }
     catch (err) {
         Bounce.rethrow(err, 'system');
-        request._setResponse(err);
```

Comment:
The one thing I could use some confirmation on is whether it's alright that up on line 23 we don't have a comparable change to this one.

I suppose now we are calling `_setResponse()` (inside `fail()`) in the case a boom error (on line 23), where in the past we were not.  In theory my guess is that this is okay because `_setResponse()` will only close the response in the case of non-boom and completed responses.  Does that sound right?  I would feel more confident if `_setResponse()` were explicitly idempotent, but I can live with this as long as someone else agrees that this is alright.

Reply:
The previous logic was hacky. There might be a slight functional change in that if the `Boom` error has a `_close` property, it could try to call it (depending on whether `this.info.completed` can be set for such a response). 

https://github.com/hapijs/hapi/blob/147b6fd14f7e03f9da1748ba09f191f154b4e7e2/lib/request.js#L528-L534

I can revert that part, as it is not essential to the fix.